### ISSUE TITLE
Error message about banned version even though it is the latest on F-Droid

### ISSUE THREAD

**Issue by Chepycou** [Helium314/SCEE#440]
Hey there,
Got repeatedly an error : “Check for banned versions keeps failing. You might run an outdated version. Please consider upgrading”. 
The problem is, this is the latest available version on F-Droid (v53.31) and I don't understand the reason for this version being banned? 

**Does it happen in normal StreetComplete?**
It does not, but in normal SC I'm running v53.2

**Versions affected**
53.31

Said message 👇️👇️👇️
![Screenshot_20230711-091310~3](https://github.com/Helium314/SCEE/assets/46412738/68c93f2b-99a4-47e1-a700-0644d9267f7e)

**Comment by Helium314** [comment]
This message shows when you try to upload, but the server with the list of banned versions can't be reached.
Did you have some internet issues trying to upload? Or maybe the list of banned versions is currently unreachable...

**Comment by Raphael124** [comment]
I was on Data and then thereafter on Wi-Fi and did not notice any issue on another app. It may be the list being unreachable. I did not test at the moment if I could upload quest data from regular SC. But if the issue arises once again, I'll try that to see if the problem also occurs on SC

**Comment by mnalis** [comment]
> Or maybe the list of banned versions is currently unreachable...

Well, there might have been issues. Does it work now?

Cogent (backbone provider where that website hosting list of banned versions for SCEE is connected) has reported two incidents on 10 and 11 July 2023 which _"might result in packet loss"_. I think those issues should be resolved by now (judging by https://ecogent.cogentco.com/network-status only some data center in New Jersey has problems now, which should not be related)

**Comment by Raphael124** [comment]
Yup, the problem does not seem to occur (I just answered a few quests to test, but the problem arose after just one quest last time). Thank you !

**Comment by Korb** [comment]
@mnalis, same with version `61.1 (6102)`:

F-Droid:
> Installed / Suggested
> Added on 02/06/2025
> Size: 103.3 MiB
> Android: 7.1+
> Requires: arm64-v8a, armeabi-v7a, x86, x86_64

[Log](https://docs.google.com/document/d/1TD6Dn9NA64irj3mi2eOPj4CdbHX-WWMvp8_9WaT-csU/edit)

Upd. `https://streetcomplete.mnalis.com/streetcomplete/banned_versions.txt` opens correctly in a browser both from a home network and from a mobile Internet connection, with and without a VPN.

> Are you a new SCEE user; or have you used SCEE before and didn't have the error, but now the error has started appearing?

I used the app for a few days, and everything seemed to work fine. Responses were sent and personal statistics changed.

> Do you have any firewall/VPN/DNS installed which might be making SCEE behave differently from your mobile web browser?
 
Yes, but I tested the app's functionality both with and without the VPN enabled. In both cases, changes are reflected on the map in the app, and in both cases, a notification appears about the version being banned.

> Could you try rebooting the phone and checking if the error is still happening?

After rebooting, the problem disappeared.

**Comment by mnalis** [comment]
> [Log](https://docs.google.com/document/d/1TD6Dn9NA64irj3mi2eOPj4CdbHX-WWMvp8_9WaT-csU/edit)

Hm, it does not seems to contain anything related. Was it taken just after the message was displayed? If so, there should be "Starting upload" message there, but I'm not seeing it in a log.

https://github.com/Helium314/SCEE/blob/4c8410cb37a69a6c080adeb60abfb31a8369e912/app/src/androidMain/kotlin/de/westnordost/streetcomplete/data/upload/Uploader.kt#L94


> Upd. `https://streetcomplete.mnalis.com/streetcomplete/banned_versions.txt` opens correctly in a browser both from a home network and from a mobile Internet connection, with and without a VPN.

OK, so that seems to tells us:
- server is working and serving file normally
- there are no network issue between user device and the server

So, the issue seems to be that SCEE cannot fetch or parse that file, while the web browser can:

https://github.com/Helium314/SCEE/blob/4c8410cb37a69a6c080adeb60abfb31a8369e912/app/src/androidMain/kotlin/de/westnordost/streetcomplete/data/upload/VersionIsBannedChecker.kt#L15-L28 

Since that file is very rarely changed, if it was about parsing, everybody would be getting that message, and it does not seem to be happening. So it seems most likely that for some reason SCEE cannot fetch that URL sometimes, but web browser can. The possible ideas that cross my mind are:

- there was a temporary network error, but SCEE somehow didn't recover from it while webbrowser did
- there was a temporary problem with system DNS, but webbrowser used DoH 1.1.1.1 or something so wasn't affected
- there is/was IPv6 networking issue, but webbrowser avoided it by using HappyEyeballs RFC8305 and falling back to IPv4 automatically, and SCEE did not

To test that last one, can you @Korb open both those URLs in mobile browser (and if not, which one is failing and with which error)?

https://ipv6.streetcomplete.mnalis.com/streetcomplete/banned_versions.txt
https://ipv4.streetcomplete.mnalis.com/streetcomplete/banned_versions.txt

Other then that, we could add extra debug logging around those checks to find why exactly it is fialing, but I don't enough Kotlin (not internals of that `httpClient`) to know how to do it...

> notification appears about the version being banned.

"About version being banned", or "about not being able to check if version is banned"? What is the message **exactly**, can you have screenshot?

> After rebooting, the problem disappeared.

And that is most interesting...  But maybe failure counter got reset to 0, so it will start showing error after about 10 new startups (i.e. app removed from recents and restarted)? Or the transient network error went away (or system chose different DNS record or different IP family to fetch it?)

**Comment by Korb** [comment]
> there was a temporary problem with system DNS, but webbrowser used DoH 1.1.1.1 or something so wasn't affected

Google Chrome uses OpenDNS, and 'Private DNS' is disabled in Android's system settings.
I usually have a VPN (without IPv6 support) enabled, which uses 1.1.1.1 and 1.0.0.1.  _(A necessary measure, because due to state censorship and boycotts in my country, the internet without a VPN is almost useless.)_

> "About version being banned", or "about not being able to check if version is banned"? What is the message exactly, can you have screenshot?

Exactly the following message (cropped at 1080×2400 px resolution and system UI scaling turned off):

> Check for banned versions keeps failing. You might run an outdated version. Please…

I deleted the screenshot when I found a problem here with exactly the same symptoms.

Upd. The next time this problem occurred, force-closing the application process and restarting it helped.
